### ISSUE TITLE
Inquiry about the Relationship Between attention_mask Length and time_steps Length

### ISSUE THREAD

**Issue by 1633347510** [appleternity/whisper_feature_extraction#1]
Whisper supports a maximum input length of 30 seconds, and the output embeddings are uniformly padded to a length of 1500. It seems that after obtaining the time steps, a further downsampling by a factor of 2 is applied to obtain the 1500-length output. However, the length of the attention mask does not change. Do you know how this is calculated?
